src/head.rs

```diff
@@ -52,6 +53,43 @@ impl Head {
     }
 }
 
+pub struct NamedHeader(pub http::header::HeaderName);
+
+pub struct Header<T>(pub T);
+
+impl<T: From<http::header::HeaderValue> + Send + 'static, S: 'static> Extract<Header<T>, S> for NamedHeader {
```

Comment:
A seed doesn't have to be a newtype; `impl Extract<Header<T>, S> for http::header::HeaderName` would work fine here! (or reversed, as of the review above.)

Reply:
I gravitated to that choice as well, but consider `String` may get confusing if used as the extractor for too many argument types. Then, consistency of providing a newtype may be more important than not nesting in a newtype. It has its pros and cons, feel free to disagree.